src/phoenix/server/api/mutations/auth.py

```diff
@@ -3,16 +3,49 @@
 from strawberry import Info
 from strawberry.permission import BasePermission
 
+from phoenix.server.bearer_auth import PhoenixUser
+
+
+class IsNotReadOnly(BasePermission):
+    message = "Application is read-only"
+
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        return not info.context.read_only
+
 
 class IsAuthenticated(BasePermission):
     message = "User is not authenticated"
 
-    async def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
-        return not info.context.read_only
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        if info.context.token_store is None:
+            return True
+        try:
+            user = info.context.request.user
+        except AttributeError:
+            return False
+        return isinstance(user, PhoenixUser) and user.is_authenticated
+
+
+class IsAdmin(BasePermission):
+    message = "Only admin can perform this action"
+
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        if info.context.token_store is None:
+            return False
+        try:
+            user = info.context.request.user
+        except AttributeError:
+            return False
+        return (
+            isinstance(user, PhoenixUser)
+            and user.is_authenticated
+            and user.claim.attributes is not None
+            and user.claim.attributes.user_role == "ADMIN"
```

Comment:
I'd assume the role would be attached to the user, not the claim attributes

Reply:
technically everything in the claim is attached to the subject, i.e. assertions about the `user`. So this is just a matter of organization: I'm mostly intending to put the non-RFC stuff into the `attributes` bucket.

Reply:
God it, so the claim is "constructed" and cached but is not a 1 to 1 reflection of what's in the JWT?

Reply:
yea, the token is basically a "signed pointer" right now, and it doesn't seem to need to contain anything else at the moment